octobot_evaluators/evaluator/strategy_evaluator.py

```diff
@@ -36,13 +47,73 @@ async def start(self, bot_id: str) -> bool:
         :return: success of the evaluator's start
         """
         await super().start(bot_id)
-        self.consumer_instance = await get_chan(MATRIX_CHANNEL, self.matrix_id).new_consumer(self.matrix_callback)
+        self.consumer_instance = await get_chan(MATRIX_CHANNEL, self.matrix_id).new_consumer(
+            self.strategy_matrix_callback)
+        return True
+
+    @classmethod
+    def get_is_requiring_complete_TA_updates(cls) -> bool:
+        """
+        :return: True if the strategy is to be waken up by technical evaluators only when every time frame update
+        has been completed. This avoids partial time frame updates wakeup.
+        """
         return True
 
     async def stop(self) -> None:
         if self.consumer_instance:
             await get_chan(MATRIX_CHANNEL, self.matrix_id).remove_consumer(self.consumer_instance)
 
+    async def strategy_matrix_callback(self,
+                                       matrix_id,
+                                       evaluator_name,
+                                       evaluator_type,
+                                       eval_note,
+                                       eval_note_type,
+                                       exchange_name,
+                                       cryptocurrency,
+                                       symbol,
+                                       time_frame):
+        # TODO: find better way than this if
+        # avoid self-calls
+        if evaluator_name == self.get_name():
+            return
+        # if this callback is from a technical evaluator: ensure strategy should be notified at this moment
+        if evaluator_type == EvaluatorMatrixTypes.TA.value:
+            # ensure this time frame is within the strategy's time frames
+            if TimeFrames(time_frame) not in self.strategy_time_frames or \
+                self.get_is_requiring_complete_TA_updates() and \
```

Comment:
`self.get_is_requiring_complete_TA_updates()` shouldn't exist in my opinion (related to comment below)